### PR TITLE
[codex] harden method-postcondition receiver stability

### PR DESCRIPTION
**Pull request by TSavo**
## Summary

- Refuse source-line-gated method postconditions when the audited method receiver is reassigned, rebound, mut-borrowed, or touched through an unknown/mutating receiver method before the audited call.
- Bound the stability scan at the audited call expression so later mutations do not over-refuse an already valid postcondition, while same-expression mutations before the call still refuse.
- Add regression tests for both sides of the boundary: pre-call reassignment/mut-borrow refuse, post-call reassignment still routes.
- Ship this as a +0 soundness-only slice. The strip_prefix +2 work is parked behind #1907 because the `method:strip_prefix` producer bridge is emitted kit-side but does not reach the final target proof.

## Verification

- `cargo test -p provekit-walk --bin provekit-walk-rpc tests::lift_implications_routes_method_postcondition_before_later_receiver_reassignment -- --exact` (RED before fix, GREEN after)
- `cargo test -p provekit-walk --bin provekit-walk-rpc tests::lift_implications_refuses_method_postcondition_after_receiver_reassignment -- --exact`
- `cargo test -p provekit-walk --bin provekit-walk-rpc tests::lift_implications_refuses_method_postcondition_after_same_expression_mut_borrow -- --exact`
- `cargo test -p provekit-walk --bin provekit-walk-rpc tests::lift_implications_matches_function_postconditions_by_arg0_path_without_fuzzy_borrowing -- --exact`
- `cargo test -p provekit-walk --bin provekit-walk-rpc tests::lift_implications_disambiguates_function_postcondition_shapes_from_manifest -- --exact`
- `cargo test -p provekit-walk --bin provekit-walk-rpc` (`159 passed; 0 failed`)
- `git diff --check`
- Pinned remote CLI cold gate with `PROVEKIT_LINKERD_BIN=/home/tsavo/remote/provekit-bcargo-d17ab89f79d4/provekit/implementations/rust/target/debug/provekit-linkerd`: `panicSafe=34`, `falsePass=0`, `silentlyDropped=0`, `droppedSites=[]`
- Pinned remote libprovekit cold gate with the same daemon pin: `panicSafe=24`, `falsePass=0`, `silentlyDropped=0`, `droppedSites=[]`

## Notes

- `platform_semantics.rs:42` remains present in the libprovekit cold output and is not dropped.
- #1906 is merged at `d901e156f5dc2982c9cdf780c82681830c12802e`.
- #1907 tracks the parked strip_prefix mint-pipeline blocker and now also has the remote stale-linkerd cold-gate infra note from this re-gate.

### CODE REVIEW

**Comment by TSavo** [comment]
@coderabbitai full review

File: implementations/rust/provekit-walk/src/bin/walk_rpc.rs [hunk 1]
```diff
@@ -3400,6 +3447,161 @@ fn expr_bare_ident_name(expr: &syn::Expr) -> Option<String> {
...
+    node: &syn::ExprMethodCall,
+    block: &syn::Block,
+    param_names: &BTreeSet<String>,
+) -> bool {
+    let Some(root) = expr_bare_ident_name(&node.receiver) else {
+        return false;
+    };
+    let occurrence = method_call_occurrence(node);
+    let stability = method_postcondition_stability_for_block(block, &occurrence);
+    if stability.effect_roots.contains(&root) {
```
[the remark this answers is not in the document]
Comment (TSavo) [reply implementations/rust/provekit-walk/src/bin/walk_rpc.rs:3495]: Fixed in 251a66e3b. The stability scan is now bounded by the full audited call expression span, so mutations after the candidate call no longer invalidate a method postcondition that applied at that earlier call.

Verification:
- `lift_implications_routes_method_postcondition_before_later_receiver_reassignment` failed before the fix and passes now.
- `lift_implications_refuses_method_postcondition_after_receiver_reassignment` still refuses mutation before the call.
- `lift_implications_refuses_method_postcondition_after_same_expression_mut_borrow` still refuses same-expression mutation before the call.
- Full `cargo test -p provekit-walk --bin provekit-walk-rpc` passed: 159/159.

**Comment by TSavo** [comment]
CodeRabbit availability fallback used for merge: the explicit `@coderabbitai full review` retry returned a bot ack, but the retained CodeRabbit review comment still reports `Review limit reached` and `Git: Failed to clone repository`, so this is treated as CodeRabbit unavailable rather than a substantive clean review.

Merge basis:
- both `prove` checks green
- Spec CID literal lint green
- macOS conformance green; remaining conformance lane nonblocking per slice-30 precedent
- CodeRabbit status context green but review unavailable as above
- Codex inline finding resolved after commit `251a66e3b`
- fresh local `cargo test -p provekit-walk --bin provekit-walk-rpc`: 159/159 passing
